sketch.js

```diff
@@ -1,137 +1,161 @@
-  console.log(shapeClassifier.data);
 }
 
-function changeLabel() {
+function changeLabel(){  // cycles through labels (see global labels)
+  // let i = 0;
   if (i < labels.length - 1){
-    // buttons.label = createButton(labels[i]);
     i++;
-    label = labels[i];
-    console.log(label, i);
+    current_label = labels[i];
   } else {
     i = 0;
-    label = labels[i];
-    console.log(label, i);
+    current_label = labels[i];
   }
-  
-}
-
-// function windowResized() {
-//   resizeCanvas(windowWidth, windowHeight);
-// }
 
-function mousePressed() {
-  // console.log("event: mousePressed()");
-  // x1 y1
-  if (keyIsDown(SHIFT) === false){
-    drawing = [];
-  }
 }
 
 function keyPressed() {
 
-  if (key === "c"){
-    drawing = [];
+  if (key === "c"){  // clears the canvas
+    DRAW = {
+      FREE:[],
+      LINE:{
+        x1:undefined,
+        y1:undefined,
+        x2:undefined,
+        y2:undefined,
+      },
+      free_store:[],
+      line_store:[]
+    }
+    // SHOULD TURN ALL OF THIS INTO A SINGLE JSON
     clear();
     background(255);
-    // BUG: Does not clear latest drawing!
-    // FIXED: draw() -> mouseDragged() + background back to white after clear
-    // BREAKS: SHIFT key line drawing!!
 
-  } else if (key === "a"){
-    shapeClassifier.addData({ image: get(0, 0, 200, 200)})
+  } else if (key === "a"){  // adds the canvas i.e. the drawn shape to the NN's data
+    img = canvas.get();
+    shapeClassifier.addData({image: img}, {label: current_label});
+
+  } else if (key === "l"){  // goes to the next label
+    changeLabel(); 
+    console.log(">>> LABEL:", current_label, "<<<");
 
+  } else if (key === "t"){  // begins training
+    shapeClassifier.train({ epochs: 50 }, finishedTraining);
   }
+
 }
 
-function mouseReleased(){
-  // console.log("event: mouseReleased()");
-  // x2 y2
-  drawing_array.push(drawing);
-  // drawing = [];
-  // 
+// DRAWING CONTROLS START HERE 
+
+function mousePressed() {
+
+  if (keyIsDown(SHIFT) === false){  // allows freehand drawing
+    DRAW.FREE = [];  // if SHIFT is not being held, start a new drawing
+  } else if (keyIsDown(SHIFT) === true) {  // allows the drawing of a straight line
+    DRAW.LINE = {x1:mouseX, y1:mouseY, x2:undefined, y2:undefined};  // CLEARS X2 AND Y2; ELSE WOULD CONNECT TO PREVIOUSLY DRAWN LINE
+  }
+
 }
 
 function mouseDragged(){
-  noFill();
-  if (mouseIsPressed){
-    drawing.push([mouseX,mouseY]);
-    beginShape();
-    for (let i = 0; i < drawing.length; i++){
-      let x = drawing[i][0];
-      let y = drawing[i][1];
-      vertex(x,y);
-    }
+
+  if (keyIsDown(SHIFT) === false){  // FREEHAND DRAWING CONTROLS
+    DRAW.FREE.push([mouseX,mouseY]); // pushing every x/y coordinate of the pressed mouse
+    noFill();
+    drawSketches(DRAW.FREE);
+
+  } else if (keyIsDown(SHIFT) === true){  // LINE DRAWING CONTROLS
+    let l = DRAW.LINE;
+    DRAW.LINE = {x1:l.x1, y1:l.y1, x2:mouseX, y2:mouseY};
```

Comment:
L108: var l is a "burner" copy of DRAW.LINE; retrieves x1 and y1
L109: x1 and y1 accessed from var l; reassigning DRAW.LINE with old (var l) and new values.

Reply:
Rename l to anchor